tests/test.py

```diff
@@ -170,21 +180,36 @@ def read_write_history_pipe():
 
 keyboard_keys = sorted(evdev.ecodes.keys.keys())[:255]
 
-fixtures = {
-    "/dev/input/event1": {
-        "capabilities": {
+
+@dataclasses.dataclass(frozen=True)
+class Fixture:
+    capabilities: Dict = dataclasses.field(default_factory=dict)
+    path: str = ""
+    name: str = "unset"
+    info: evdev.device.DeviceInfo = evdev.device.DeviceInfo(None, None, None, None)
+    phys: str = "unset"
+    group_key: Optional[str] = None
+
+    def __hash__(self):
+        return hash(self.path)
+
+
+class _Fixtures:
```

Comment:
Docstring please.

It seems `_Fixtures` is some sort of container. Is it like a dict?

Reply:
It is just a class with a bunch of class variables.
It defines all the standard fixtures and provides access to them via the path (`fixtures["/dev/input/eventX"]`) the attribute name (`fixtures.dev_input_eventX`) and a property name (`fixtures.foo_device_2_mouse`).
It also allows dynamic creation of new fixtures during runtime.